dlog/logger.go

```diff
@@ -84,6 +84,19 @@ type OptimizedLogger interface {
 	UnformattedLogf(level LogLevel, format string, args ...interface{})
 }
 
+// LoggerWithMaxLevel can be implemented by loggers that define a maximum
+// level that will be logged, i.e. if a logger defines a max-level of
+// LogLevelInfo, then only LogLevelError, LogLevelWarn, and LogLevelInfo will
+// be logged, and LogLevelDebug and LogLevelTrace will be discarded.
```

Comment:
nit: since this comma is delimiting a set of comma-separated lists (`LogLevelError, LogLevelWarn, and LogLevelInfo` vs `LogLevelDebug and LogLevelTrace`), it should be a semicolon to indicate that it is higher precedence than the other commas.